addons/project/tests/test_task_dependencies.py

```diff
@@ -159,3 +159,76 @@ def set_task_dependencies_setting(enabled):
             'name': 'My Ducks Project'
         })
         self.assertFalse(self.project_ducks.allow_task_dependencies, "New Projects allow_task_dependencies should default to group_project_task_dependencies")
+
+    def test_duplicate_project_with_task_dependencies(self):
+
+        self.project_pigs.allow_task_dependencies = True
+        self.task_1.depend_on_ids = self.task_2
+        pigs_copy = self.project_pigs.copy()
+
+        task1_copy = pigs_copy.task_ids.filtered(lambda t: t.name == 'Pigs UserTask')
+        task2_copy = pigs_copy.task_ids.filtered(lambda t: t.name == 'Pigs ManagerTask')
+
+        self.assertEqual(len(task1_copy), 1, "Should only contain 1 copy of UserTask")
+        self.assertEqual(len(task2_copy), 1, "Should only contain 1 copy of ManagerTask")
+
+        self.assertEqual(task1_copy.depend_on_ids.ids, [task2_copy.id],
+                         "Copy should only create a relation between both copy if they are both part of the project")
+
+        task1_copy.depend_on_ids = self.task_1
+
+        pigs_copy_copy = pigs_copy.copy()
+        task1_copy_copy = pigs_copy_copy.task_ids.filtered(lambda t: t.name == 'Pigs UserTask')
+
+        self.assertEqual(task1_copy_copy.depend_on_ids.ids, [self.task_1.id],
+                         "Copy should not alter the relation if the other task is in a different project")
+
+    def test_duplicate_project_with_subtask_dependencies(self):
+
+        parent_task = self.env['project.task'].with_context({'mail_create_nolog': True}).create({
+            'name': 'Parent Task',
+            'project_id': self.project_goats.id,
+            'child_ids': [
+                Command.create({'name': 'Node 1'}),
+                Command.create({'name': 'SuperNode 2', 'child_ids': [Command.create({'name': 'Node 2'})]}),
+                Command.create({'name': 'Node 3'}),
+            ],
+        })
+
+        node1 = parent_task.child_ids[0]
+        node2 = parent_task.child_ids[1].child_ids[0]
```

Comment:
Since there is only one child in the subtask, it is not necessarily needed to take the first element in the recordset contained in the `child_ids` since the recordset contains only one task as I said above. So, the following suggestion is equivalent:
```suggestion
        node2 = parent_task.child_ids[1].child_ids
```
🙂